viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,58 @@
-    nearPlane.setFromNormalAndCoplanarPoint(cameraDirection, nearPlaneCoplanarPoint);
-    let far = -Infinity;
-    for (let i = 0; i < corners.length; ++i) {
-      if (nearPlane.distanceToPoint(corners[i]) >= 0) {
-        const dist = corners[i].distanceTo(cameraPosition);
-        far = Math.max(far, dist);
-      }
-    }
-    far = Math.max(near * 2, far);
-
-    return far;
-  }
-
-  private calculateCameraNear(
-    camera: THREE.PerspectiveCamera,
-    combinedBbox: THREE.Box3,
-    cameraPosition: THREE.Vector3
-  ): number {
-    let near = combinedBbox.distanceToPoint(cameraPosition);
-    near /= Math.sqrt(1 + Math.tan(((camera.fov / 180) * Math.PI) / 2) ** 2 * (camera.aspect ** 2 + 1));
-    near = Math.max(0.1, near);
-
-    return near;
-  }
+  setCameraState(state: CameraState): void;
 
   /**
-   * Calculates new target when raycaster doesn't have any intersections with the model.
-   * @param cursorPosition Cursor position for desired calculations.
-   * @param cursorPosition.x
-   * @param cursorPosition.y
+   * Get camera's state
    */
-  private calculateNewTargetWithoutModel(
-    cursorPosition: { x: number; y: number },
-    modelsBoundingBox: THREE.Box3
-  ): THREE.Vector3 {
-    const modelSize = modelsBoundingBox.min.distanceTo(modelsBoundingBox.max);
-
-    const lastScrollTargetDistance = this.cameraControls.getScrollTarget().distanceTo(this._camera.position);
-
-    const newTargetDistance =
-      lastScrollTargetDistance <= this.cameraControls.minDistance
-        ? Math.min(this._camera.position.distanceTo(modelsBoundingBox.getCenter(new THREE.Vector3())), modelSize) / 2
-        : lastScrollTargetDistance;
-
-    this._raycaster.setFromCamera(cursorPosition, this._camera);
-
-    const farPoint = this._raycaster.ray.direction
-      .clone()
-      .normalize()
-      .multiplyScalar(newTargetDistance)
-      .add(this._camera.position);
-
-    return farPoint;
-  }
+  getCameraState(): Required<CameraState>;
 
   /**
-   * Calculates new camera target based on cursor position.
-   * @param event MouseEvent that contains pointer location data.
+   * Required for understadning of `cameraChanged` definition for other parts of Reveal
+   * @param event
+   * @param callback
    */
-  private async calculateNewTarget(event: MouseEvent): Promise<THREE.Vector3> {
-    const { offsetX, offsetY } = event;
-    const { x, y } = this.convertPixelCoordinatesToNormalized(offsetX, offsetY);
-
-    const modelRaycastData = await this._modelRaycastCallback(offsetX, offsetY);
-
-    const newTarget =
-      modelRaycastData.intersection?.point ??
-      this.calculateNewTargetWithoutModel({ x, y }, modelRaycastData.modelsBoundingBox);
-
-    return newTarget;
-  }
-
+  on(event: 'cameraChange', callback: CameraChangeData): void;
   /**
-   * Removes controls event listeners if they are defined.
+   * Required for understadning of `cameraChanged` definition for other parts of Reveal
+   * @param event
+   * @param callback
```

Comment:
Change to something about "unsubscribe to event", current description doesn't say what this function does - it says what the event does (which doesn't make sense in the unsubscribe function)